src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +698,45 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  addAuthorToResponse(qr, practitionerRef) {
+    function traverseToItemsLeafNode(item, practitionerRef) {
+      if (!item.item) {
+        return addAuthor(item, practitionerRef);
+      }
+      else {
+        item.item.map(item => {
+          traverseToItemsLeafNode(item, practitionerRef);
+        })
+      }
+    }
+    // url is a string
+    function addAuthor(item, practitionerRef) {
+      var url = "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author"
+      const urlValRef =
+      {
+        "url": url,
+        "valueReference":
+        {
+          "reference": practitionerRef
+        }
+      }
+      if (item.extension) {
+        item.extension.push(urlValRef)
+      }
+      else {
+        item["extension"] = [urlValRef]
+        console.log(item);
+        console.log(practitionerRef);
```

Comment:
Are these console log statement only meant for the else case? Also, are those logging statements used only for debugging or needed for actual application? If it is for the former only, then we should remove them.